snoop/common_data/models.py

```diff
@@ -36,3 +41,52 @@ class Meta:
     class PartitioningMeta:
         method = PostgresPartitioningMethod.HASH
         key = ["doc_sha3_256"]
+
+
+class NextcloudCollection(models.Model, Collection):
+    """Model for storing nextcloud collection metadata."""
+
+    name = models.CharField(max_length=256, unique=True)
+    opt = models.JSONField(null=True, blank=True)
+
+    @property
+    def webdav_user(self):
+        return self.opt.get('webdav_user', '')
+
+    @property
+    def webdav_password(self):
+        return self.opt.get('webdav_password', '')
+
+    @property
+    def webdav_url(self):
+        return self.opt.get('webdav_url', '')
+
+    @contextmanager
+    def mount_collection_webdav(self, col, nc_col):
+        """Mount a nextcloud collection via webdav.
+        """
+        subprocess.run(['mkdir', '-p',
+                        (
+                            f'{settings.SNOOP_WEBDAV_MOUNT_DIR}'
+                            f'/{self.name}/data'
+                        )
+                        ], check=True)
+        secrets_content = (
+            f'{settings.SNOOP_WEBDAV_MOUNT_DIR}'
+            f'/{self.name}/data'
+            f' {self.webdav_user} {self.webdav_password}'
+        )
+        with open('/etc/davfs2/secrets', 'a') as secrets_file:
```

Comment:
1. if you open this in append mode, how do you know you won't get duplicate entries every time you run this function?
2. if this gets executed in parallel by many workers, how do you know this file won't be corrupted?
3. can we move this into a separate file for each collection, that you put a file lock around?
4. is there a method for removing old/unused mounts? We don't want to have a million active mounts around all the time